crates/nu-command/src/filesystem/save.rs

```diff
@@ -318,10 +322,20 @@ fn stream_to_file(
     mut stream: RawStream,
     file: File,
     span: Span,
+    progress: bool,
 ) -> Result<PipelineData, ShellError> {
     let mut writer = BufWriter::new(file);
 
-    stream
+    let mut bytes_processed: u64 = 0;
+    let bytes_processed_p = &mut bytes_processed;
+    let file_total_size = stream.known_size.map(|n| (n as u64)); // type TBD
+
+    // Create the progress bar and set the style.
+    let mut bar = progress_bar::NuProgressBar::new(file_total_size);
```

Comment:
Would it make sense to avoid creating the progress bar when `--progress` is not set?

I'm concerned that it might write something to stdout when `--progress` is not set, or be slow to initialize in scripts that save a lot of files.

Reply:
Good point. I though about it but because I call `bar` from inside `stream.try_for_each` I though it was going to get messy. I only think about two possible solutions:
1. we write `stream.try_for_each` twice. One with the bar and another one without it. but this will make code messy and the executable bigger. don't think its the best
2. We make `bar` an Option<NuProgressBar> ?

If you know a better way I'll implement it :)

Reply:
@rgwood  What do you think about 
```rust
// Create the progress bar and set the style.
    let mut bar = if progress {
        Some(progress_bar::NuProgressBar::new(file_total_size))
    } else {
        None
    };
```

And check the `progress` flag inside the stream to update the progress bar,
 and before the final return to print the finished msg

Reply:
Sounds good 👍